backend/src/users/controllers.ts

```diff
@@ -186,24 +186,36 @@ const getUsers = async (
     },
   });
 
-  const queryResult = await prisma.user.findMany({
-    where: {
-      AND: [whereDict],
-    },
-    include: {
-      profile: true,
-      events: eventId ? true : false,
-    },
+  let queryOptions = {
+    where: { AND: [whereDict] },
+    include: { profile: true, events: eventId ? true : false },
     orderBy: sortDict[sort.key],
     take: take,
     skip: skip,
-    cursor: cursor,
-  });
+  };
+  let queryResult;
+  if (cursor) {
+    queryResult = await prisma.user.findMany({
+      ...queryOptions,
+      cursor,
+    });
+  } else {
+    queryResult = await prisma.user.findMany({
+      ...queryOptions,
+    });
+  }
+
   const lastPostInResults = take
     ? queryResult[take - 1]
     : queryResult[queryResult.length - 1];
-  const myCursor = lastPostInResults ? lastPostInResults.id : undefined;
-  return { result: queryResult, cursor: myCursor, totalItems: totalRecords };
+  const nextCursor = lastPostInResults ? lastPostInResults.id : undefined;
+  const prevCursor = queryResult[0] ? queryResult[0].id : undefined;
+  return {
+    result: queryResult,
+    nextCursor: nextCursor,
+    prevCursor: prevCursor,
+    totalItems: totalRecords,
+  };
```

Comment:
New design here:

Based on cursor pagination. NextCursor is the id of the last user returned for this batch. We will use this to fetch the next batch (basically users after this id in the db). PrevCursor is the id of the first user in the batch. So if we need to fetch a previous page, we just take the users before that id

Reply:
This also means that all components that used to consume this endpoint will break because what this returns has changed